### PR TITLE
Dynamic arrays

### PR DESCRIPTION
**Pull request by dwalintukan**

### CODE REVIEW

File: src/contract.js [hunk 1]
```diff
@@ -88,49 +80,75 @@ class Contract {
...
-          dataHex = dataHex.concat(hex);
+    _.each(methodObj.inputs, (item, index) => {
+      const type = item.type;
+      let hex;
+
+      if (type === Constants.BYTES
+        || type === Constants.STRING
+        || type.match(Constants.REGEX_DYNAMIC_ARRAY)) { // dynamic types
+        let data = '';
+        if (type === Constants.BYTES) {
```
Comment (frankobe) [inline src/contract.js]: why not throw error directly using one if

File: src/contract.js [hunk 2]
```diff
@@ -88,49 +80,75 @@ class Contract {
...
+          console.error('dynamics bytes conversion not implemented.');
+        } else if (type === Constants.STRING) {
```
Comment (frankobe) [inline src/contract.js]: why not throw error directly using one if

File: src/encoder.js [hunk 3]
```diff
@@ -4,10 +4,65 @@ const BigNumber = require('bignumber.js');
...
+      hex = this.stringToHex(value, Constants.MAX_HEX_CHARS_PER_BYTE);
+    } else if (type.match(Constants.REGEX_STATIC_BYTES_ARRAY)) { // fixed bytes array, ie. bytes32[10]
+      const arrCapacity = _.toNumber(type.match(Constants.REGEX_NUMBER)[1]);
+      if (value instanceof Array) {
+        hex = this.stringArrayToHex(value, arrCapacity);
+      } else {
+        hex = this.stringToHex(value, Constants.MAX_HEX_CHARS_PER_BYTE * arrCapacity);
+      }
+    } else {
+      console.error(`Unimplemented type: ${type}`);
```
Comment (frankobe) [inline src/encoder.js]: so we dunt support encoding of dynamic array?

File: src/contract.js (hunk 1 above)
Comment (dwalintukan) [reply src/contract.js]: ya better to throw since their datahex wont be right anyways. 👍

File: src/contract.js (hunk 2 above)
Comment (dwalintukan) [reply src/contract.js]: changed

File: src/encoder.js (hunk 3 above)
Comment (dwalintukan) [reply src/encoder.js]: We do. L129 of `contract.js` uses `Encoder.encodeParam` method. When we send it from contract.js, we are encoding it as the array type individually. For code reusability.

**Pull request merged**